RitoxicGame/space-trader
# Qp trade UI -- what I have so far

## What did you add in this branch?

Trade UI Beginnings:
Basic framework for a (scuffed) trade UI
Framework for commodities
Buttons to tie everything together
Lots and lots of scripts
Update: now _actually_ ready to test!
## How can we test these changes?

Go into the Game scene and play.
Click the Trade buttons in the top-right.
There should a window that appears in the middle of the screen with some panels representing the resources available on that planet (right now everything has the icon for Ore, so it might be a bit confusing).
If not, let me know as I might have messed up the scripts in a prior commit.

Go into the trade menus and fiddle around: can you break anything?
Do the buttons line up with each other? Is it at least somewhat readable?
Are there any obvious mistakes in formatting or math?
Do all of the menus toggle on and off when they're supposed to?
Do any of the text boxes overlap with each other?

NOTE: a known issue with GridLayoutGroups is their lack of flexibility when it comes to resolution scaling. What this means in our context is that our current trade menu tiles **do not** scale well with resolution: they will become oddly big in high resolutions and very small and deformed in low resolutions. For the time being, just ensure that all resolutions are kept to a consistent 16:9 / 1920x1080.

## Relevant screenshots (if applicable):

![Screenshot 2024-03-25 001931](https://github.com/RitoxicGame/space-trader/assets/48595591/49344a43-9083-4d24-9c4b-b072c5de31a1)
![Screenshot 2024-03-25 001937](https://github.com/RitoxicGame/space-trader/assets/48595591/6180842c-2025-4097-bf70-2a5d0c19b6ed)
![Screenshot 2024-03-25 001948](https://github.com/RitoxicGame/space-trader/assets/48595591/5dc889b3-6465-47a5-9902-22a2d218b629)
![Screenshot 2024-03-25 002000](https://github.com/RitoxicGame/space-trader/assets/48595591/d1db7e54-7966-4f91-96c3-8a6822e0bc32)

## What week is this due by?

Alpha/Playtest 1
